pennylane_lightning/core/src/simulators/lightning_kokkos/StateVectorKokkos.hpp

```diff
@@ -257,96 +257,62 @@ class StateVectorKokkos final
     }
 
     /**
-     * @brief Apply a single qubit operator to the state vector using a matrix
+     * @brief Apply a multi qubit operator to the state vector using a matrix
      *
      * @param matrix Kokkos gate matrix in the device space
      * @param wires Wires to apply gate to.
      * @param inverse Indicates whether to use adjoint of gate.
      */
-    void applySingleQubitOp(const KokkosVector &matrix,
-                            const std::vector<size_t> &wires,
-                            bool inverse = false) {
+    void applyMultiQubitOp(const KokkosVector &matrix,
+                           const std::vector<std::size_t> &wires,
+                           bool inverse = false) {
         auto &&num_qubits = this->getNumQubits();
-        if (!inverse) {
+        std::size_t two2N = std::exp2(num_qubits - wires.size());
+        std::size_t dim = std::exp2(wires.size());
+        KokkosVector matrix_trans("matrix_trans", matrix.size());
+
+        if (inverse) {
+            Kokkos::MDRangePolicy<Kokkos::Rank<2>> policy_2d({0, 0},
```

Comment:
Can we use a const variable instead of `2` here?

Reply:
This is replacing a double `parallel_for` with `RangePolicy`s, so we could (maybe) have a constant like `constexpr int double_loop_rank=2`, but I don't see the benefit in this case.

Reply:
It would be nice to name it as `double_loop_rank` for the maintenance purpose.